File: programers_Lv1/등차수열의 특정한 항만 더하기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181931?language=javascript
function solution(a, d, included) {
  return included
    .map((el, idx) => (el ? a + idx * d : 0))
    .reduce((acc, curr) => {
      return (acc += curr);
    }, 0);
}

console.log(solution(3, 4, [true, false, false, true, true]));

File: programers_Lv1/등차수열의 특정한 항만 더하기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181931?language=python3
def solution(a, d, included):
    return sum(a + i*d for i, t in enumerate(included) if t)

print(solution(3, 4, [True, False, False, True, True]))
